src/utils/task-run-filter-transforms.ts

```diff
@@ -0,0 +1,89 @@
+import { PipelineRunLabel } from '~/consts/pipelinerun';
+import { MatchExpression, Selector, WatchK8sResource } from '~/types/k8s';
+
+/**
+ * Extended selector type that includes typed filter fields for TaskRuns
+ */
+export type TaskRunSelector = Selector &
+  Partial<{
+    filterByName: string;
+    filterByCreationTimestampAfter: string;
+    filterByCommit: string;
+  }>;
+
+/**
+ * Converts a TaskRunSelector (Tekton-style filter object) into a format
+ * suitable for KubeArchive resource queries.
+ *
+ * Specifically:
+ * - Converts `filterByName` into a Kubernetes field selector string (`metadata.name=...`).
+ * - Converts `filterByCommit` into a label-based matchExpression.
+ * - Preserves any existing `matchLabels` and `matchExpressions`.
+ * - Returns an object containing:
+ *   - `selector`: a Selector object with labels and expressions for KubeArchive.
+ *   - `fieldSelector`: a pre-built string suitable for K8s API field selectors.
+ *
+ * Notes:
+ * - `filterByCreationTimestampAfter` cannot be expressed as a K8s fieldSelector
+ *   (Kubernetes does not support comparison operators in field selectors). This
+ *   filter must be applied client-side.
+ *
+ * @param filterBy - TaskRunSelector object containing Tekton-style filter options.
+ * @returns An object with `selector` and `fieldSelector` ready for use with KubeArchive hooks.
+ */
+export const convertFilterToKubearchiveSelectors = (
+  filterBy: TaskRunSelector,
+): Pick<WatchK8sResource, 'fieldSelector' | 'selector'> => {
+  const fieldSelectors: Record<string, string> = {};
+  if (filterBy.filterByName) fieldSelectors['metadata.name'] = filterBy.filterByName;
+
+  const fieldSelector = Object.keys(fieldSelectors).length
+    ? Object.entries(fieldSelectors)
+        .map(([k, v]) => `${k}=${v}`)
+        .join(',')
+    : undefined;
+
+  // Build matchExpressions (including commit filter)
+  const matchExpressions: MatchExpression[] = [
+    ...(filterBy.matchExpressions ?? []),
+    ...(filterBy.filterByCommit
+      ? [
```

Comment:
nit: we have a `createEquals` utility func in the `src/k8s/k8s-utils.ts` file - I'm wondering if it could be beneficial to define it somewhere else and reuse in both places 🤔